scripts/checks/extract_scorecard_artifact.py

```
@@ -0,0 +1,106 @@
+import zipfile
+from pathlib import Path
+
+EXPECTED_MEMBER = "results.sarif"
+
+
+def resolve_artifact_zip(source: Path) -> Path:
+    """Return the artifact ZIP file from a file path or single-ZIP directory."""
+    if source.is_file():
+        return source
+    if not source.is_dir():
+        raise ValueError(f"artifact source does not exist: {source}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)